projectq/backends/_ibm/_ibm.py

```diff
@@ -62,6 +63,11 @@ def __init__(self, use_hardware=False, num_runs=1024, verbose=False,
             password (string): IBM Quantum Experience password
             device (string): Device to use ('ibmqx4', or 'ibmqx5')
                 if use_hardware is set to True. Default is ibmqx4.
+            num_retries (int): Number of times to retry to obtain
+                results from the IBM API. (default is 3000)
+            interval (int): Number of seconds between successive
+                attempts to obtain results from the IBM API.
+                (default is 1)
```

Comment:
I wouldn't make this an `int`. We might even only expose `num_retries` and drop `interval` here; what do you think?

Reply:
True, the (int) is not necessary. Fixed in 53d2483. But why not expose `interval`? Gives your users more freedom and I don't see any downsides. In fact, it makes it more obvious that the API works by periodically polling the IBM servers for a result, which is a nice side effect.

Reply:
Because I don't see why I'd ever want to change the interval; do you?
Regarding your reason for adding it: One can also just add a comment in the docs about the "periodically polling the IBM servers" to make it obvious.

Reply:
What makes you so sure that, say 1s, is always optimal for any application using ProjectQ? Maybe someone needs a result as soon as it is available and cannot wait for at least 1s, and maybe someone else knows that their computation will take a while and they don't want to spam IBM with requests to their API that they anyway know will yield no result.

Reply:
I didn't claim it was optimal; I was just asking if you saw a use case.

If I cannot afford to wait for 1s, then I wouldn't use the service in the first place since the queue tends to be quite long. If I want to retrieve the results at a later point (e.g., because I know that the queue is too long), I'd just use `retrieve_execution`.

Since you find it useful to have access to `interval` (beyond it having no downsides), we should add it.

Reply:
My perspective is that of someone developing a library on top of ProjectQ (https://pennylane-pq.readthedocs.io/en/latest/installation.html) and and in my code I anyway have to block until I get results from the queue because the following code immediately depends on the outcome. Using retreive_execution is thus not a good option, but having an option to have the user set `interval` would be nice :-)